llvm/lib/TableGen/TGLexer.cpp

```diff
@@ -834,6 +830,24 @@ tgtok::TokKind TGLexer::lexPreprocessor(
       return tgtok::Error;
     }
 
+    return LexToken();
+  } else if (Kind == tgtok::Undef) {
+    StringRef MacroName = prepLexMacroName();
+    if (MacroName.empty())
+      return ReturnError(TokStart, "Expected macor name after #undef");
+
+    if (!DefinedMacros.erase(MacroName))
+      return ReturnError(TokStart, "undefine(#undef) an undefined macro");
```

Comment:
In the C preprocessor, this case isn't an error. That may come as a surprise to folks trying to use this.